azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworkPeeringsImpl.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.network.implementation;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.apigeneration.LangDefinition;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.NetworkPeering;
+import com.microsoft.azure.management.network.NetworkPeerings;
+import com.microsoft.azure.management.resources.fluentcore.arm.ResourceUtils;
+import com.microsoft.azure.management.resources.fluentcore.arm.collection.implementation.IndependentChildrenImpl;
+
+import rx.Completable;
+import rx.Observable;
+import rx.functions.Func1;
+
+/**
+ *  Implementation for network peerings.
+ */
+@LangDefinition
+class NetworkPeeringsImpl
+    extends IndependentChildrenImpl<
+        NetworkPeering,
+        NetworkPeeringImpl,
+        VirtualNetworkPeeringInner,
+        VirtualNetworkPeeringsInner,
+        NetworkManager,
+        Network>
+
+    implements NetworkPeerings {
+
+    private final NetworkImpl network;
+
+    // Constructor to use from the context of a parent
+    NetworkPeeringsImpl(final NetworkImpl parent) {
+        super(parent.manager().inner().virtualNetworkPeerings(), parent.manager());
+        this.network = parent;
+    }
+
+    @Override
+    public NetworkPeeringImpl define(String name) {
+        return wrapModel(name);
+    }
+
+    // Fluent model create helpers
+
+    @Override
+    protected NetworkPeeringImpl wrapModel(String name) {
+        VirtualNetworkPeeringInner inner = new VirtualNetworkPeeringInner().withName(name);
+        return new NetworkPeeringImpl(inner, this.network);
+    }
+
+    @Override
+    protected NetworkPeeringImpl wrapModel(VirtualNetworkPeeringInner inner) {
+        return (inner != null) ? new NetworkPeeringImpl(inner, this.network) : null;
+    }
+
+    @Override
+    public Completable deleteByParentAsync(String groupName, String parentName, final String name) {
+        return this.manager().networks()
+            // Get the parent network of the peering to delete
+            .getByResourceGroupAsync(groupName, parentName)
+
+            // Find the local peering to delete
+            .flatMap(new Func1<Network, Observable<NetworkPeering>>() {
+                @Override
+                public Observable<NetworkPeering> call(Network localNetwork) {
+                    if (localNetwork == null) {
+                        return Observable.empty(); // Missing local network, so nothing else to do
```

Comment:
Since `Observable.empty()` is returned in some cases, none of the down-stream flatmap will be called, and the emitted sequence is value-less. The final `last()` will be applied on value-less stream and which throws "java.util.NoSuchElementException: Sequence contains no elements".

Simple repro is:

```java
.flatMap(new Func1<Object, Observable<Integer>>() {
        @Override
        public Observable<Integer> call(Object o) {
            // This will not be called since there is no onNext call in upstream
            //
            return null;
        }
    })
    .last()
    .toCompletable()
    .await();   // Throws java.util.NoSuchElementException: Sequence contains no elements
```

Reply:
This can be solved if we special handle the empty case by applying `switchIfEmpty()` operator before `.last()`.

```java
.flatMap(new Func1<Object, Observable<Integer>>() {
        @Override
        public Observable<Integer> call(Object o) {
            // This will not be called since there is no onNext call in upstream
            //
            return null;
        }
    })
    .switchIfEmpty(Observable.<Integer>just(null))
    .last()
    .toCompletable()
    .await();   
```

In your case `.switchIfEmpty(Observable.<Void>just(null))`

Reply:
@anuchandy wouldn't it be simpler to just return `Observable.just(null)` instead of `Observable.empty()` here too though, since the rest of the downstream can handle this output being null?

Reply:
Yes, we can also handle this by returning observable.just(null)..